frontend/src/stimulus/controllers/async-dialog.controller.ts

```diff
@@ -34,10 +34,20 @@ import { TurboHelpers } from '../../turbo/helpers';
 
 export default class AsyncDialogController extends ApplicationController {
   connect() {
-    this.element.addEventListener('click', (e) => {
-      e.preventDefault();
+    this.element.addEventListener('click', (event:MouseEvent) => this.handleClick(event));
+    this.element.addEventListener('keydown', (event:KeyboardEvent) => this.handleKeydown(event));
+  }
+
+  handleClick(event:MouseEvent) {
+    event.preventDefault();
+    this.triggerTurboStream();
+  }
```

Comment:
Inline with the recommended [Stimulus naming conventions](https://stimulus.hotwired.dev/reference/actions#naming-conventions), perhaps we can rename this something like `showDialog`?

```suggestion
  openDialog(event:Event) {
```